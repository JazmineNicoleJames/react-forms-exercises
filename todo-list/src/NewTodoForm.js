import React, {useEffect, useState} from "react";


const NewTodoForm = ({addTodo, editTodo, updateTodo}) => {

    const initialState = {
        id:"",
        todo:""
    };
    const [formData, setFormData] = useState(initialState);
    console.log('formData', formData)
    useEffect(() => {
        if(editTodo !== null && editTodo !== undefined){
            console.log('editTodo', editTodo)
            setFormData({...editTodo, todo: editTodo.todo || ""})
        } else {
            setFormData(initialState)
        }
    }, [editTodo])


    const handleSubmit = (e) => {
        e.preventDefault();
        if(editTodo !== null){
            updateTodo(formData)
    
        } else {
            addTodo(formData)
            setFormData(initialState)
        }
        setFormData(initialState)
    }

    const handleChange = evt => {
        const {name, value} = evt.target;
        setFormData({
            ...formData,
            [name]:value
        })
    };

    return (
        <form data-testid="new-todo-form" className="NewTodoForm">
            <label htmlFor="todo">Todo:</label>
            <input type="text" 
                    id="todo"
                    placeholder="todo" 
                    name="todo"
                    value={formData.todo}
                    onChange={handleChange} />
            <button onClick={handleSubmit}>{editTodo ? "Update todo" : "Add todo"}</button>
        </form>
    );
};

export default NewTodoForm;