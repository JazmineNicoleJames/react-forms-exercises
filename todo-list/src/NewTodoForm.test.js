import NewTodoForm from "./NewTodoForm";
import { render, fireEvent} from "@testing-library/react";


it("should have todo form", () => {
    const { getByTestId } = render(<NewTodoForm />)
    const newTodo = getByTestId('new-todo-form');
    console.log('newTodo', newTodo)
    expect(newTodo).toBeInTheDocument();
});

it("should match snapshot", () => {
    const { asFragment } = render(<NewTodoForm />);

    expect(asFragment).toMatchSnapshot();
})