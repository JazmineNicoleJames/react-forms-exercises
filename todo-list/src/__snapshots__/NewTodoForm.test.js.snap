// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have todo form 1`] = `[Function]`;

exports[`should match snapshot 1`] = `[Function]`;
